testslide/matchers.py

```diff
@@ -227,31 +251,55 @@ def __init__(self) -> None:
 
 class NotThisFloat(_FloatComparison):
     def __init__(self, ne: float) -> None:
+        if not isinstance(ne, float):
+            raise ValueError(
+                f"NotThisFloat(...) expects a 'float' as argument while '{type(ne).__name__}' was provided"
+            )
         super().__init__(ne=ne)
 
 
 class FloatBetween(_FloatComparison):
     def __init__(self, lower: float, upper: float) -> None:
+        if not isinstance(lower, float) or not isinstance(upper, float):
```

Comment:
I would allow integers values as well.
The same reasoning of `IntBetween` holds in this case.

Let's say that I want to match any number that is between 0 and 100 ... why should I be forced to represent them as floats?